third_party/move/move-compiler-v2/src/file_format_generator/module_generator.rs

```diff
@@ -418,11 +429,34 @@ impl ModuleGenerator {
             loc,
             fun_env.get_result_type().flatten().into_iter().collect(),
         );
-        let access_specifiers = fun_env.get_access_specifiers().as_ref().map(|v| {
-            v.iter()
-                .map(|s| self.access_specifier(ctx, fun_env, s))
-                .collect()
-        });
+        let access_specifiers = if self.gen_access_specifiers {
+            fun_env.get_access_specifiers().as_ref().map(|v| {
+                v.iter()
+                    .map(|s| self.access_specifier(ctx, fun_env, s))
+                    .collect()
+            })
+        } else {
+            // Report an error if we cannot drop the access specifiers.
+            // TODO(#12623): remove this once the bug is fixed
+            if fun_env
+                .get_access_specifiers()
+                .map(|v| {
+                    v.iter().any(|s| {
+                        s.kind != AccessKind::Acquires
```

Comment:
This looks similar to `AccessSpecifier::is_old_style_acquires()`. Use that instead?

Reply:
Unfortunately, that one is on the file_format version of the thing, whereas here we look at the AST with location info. This will go away, so also no point to factor it out.